opts/gpus.go

```diff
@@ -0,0 +1,95 @@
+package opts
+
+import (
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/docker/docker/api/types/container"
+)
+
+// GpuOpts is a Value type for parsing mounts
+type GpuOpts struct {
+	values []container.DeviceRequest
+}
+
+// Set a new mount value
+// nolint: gocyclo
+func (o *GpuOpts) Set(value string) error {
+	csvReader := csv.NewReader(strings.NewReader(value))
+	fields, err := csvReader.Read()
+	if err != nil {
+		return err
+	}
+
+	req := container.DeviceRequest{Options: make(map[string]string), Capabilities: [][]string{{"gpu"}}}
+
+	// Set writable as the default
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		key := strings.ToLower(parts[0])
+
+		if len(parts) == 1 {
+			req.Count = key
+			continue
+		}
+
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid field '%s' must be a key=value pair", field)
+		}
+
+		value := parts[1]
+		switch key {
+		case "driver":
+			req.Driver = value
+		case "count":
+			req.Count = value
+		case "device":
+			req.DeviceIDs = strings.Split(value, ",")
```

Comment:
@RenaudWasTaken for device IDs, I don't want to use `id` because I want to be consistent with other future plans (binding `id` to a "gpu request" instead, providing a more secure API which is needed for swarm and buildkit), so I went with `device`.

Reply:
I'm ok with this :)